### PR TITLE
Feature/797 create smart contract

### PR DESCRIPTION
**Pull request by biscuitdey**
# Description

This PR adds a solidity contract to store the anchor hashes for each workgroupId.

## Related Issue

#797 

## Motivation and Context

Store anchor hashes (merkle tree root + witness) on the ccsm

## How Has This Been Tested

Tests would be added in upcoming PRs.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Request to be added as a Code Owner/Maintainer

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [ ] I commit to abide by the Responsibilities of Code Owners/Maintainers.

### CODE REVIEW

File: examples/bri-3/src/bri/ccsm/contracts/Ccsm.sol [hunk 1]
```diff
@@ -0,0 +1,25 @@
+//SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.8.17;
+
+contract Ccsm {
```
Comment (Therecanbeonlyone1969) [inline examples/bri-3/src/bri/ccsm/contracts/Ccsm.sol:4]: name should probably be `ccsm-bpi-state-anchor`

Comment (Therecanbeonlyone1969) [reply examples/bri-3/src/bri/ccsm/contracts/Ccsm.sol:4]: the contract needs access control:

```
import "@openzeppelin/contracts/access/AccessControl.sol";

contract MyContract is AccessControl {

  bytes32 public constant ADMIN_ROLE = keccak256("ADMIN_ROLE");

  constructor(address[] memory admins) {
    _setupRole(DEFAULT_ADMIN_ROLE, msg.sender); // Grant deployer the default admin role

    for (uint i = 0; i < admins.length; i++) {
      _grantRole(ADMIN_ROLE, admins[i]); // Grant admin role to each address
    }
  }

  // Your contract functions here

  modifier onlyAdmin() {
    require(hasRole(ADMIN_ROLE, msg.sender), "Only admin can call this function");
    _;
  }
```

and the change in `setAnchorHash` is 
```
function setAnchorHash(
    string calldata _workgroupId,
    string calldata _anchorHash
  ) external onlyAdmin
```

File: examples/bri-3/src/bri/ccsm/contracts/Ccsm.sol [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+  mapping(string => string) public anchorHashStore;
+  event AnchorHashSet(string indexed workgroupId, string anchorHash);
+
+  function setAnchorHash(
+    string calldata _workgroupId,
+    string calldata _anchorHash
+  ) external {
+    require(bytes(_workgroupId).length > 0, 'WorkgroupId cannot be empty');
```
Comment (Therecanbeonlyone1969) [inline examples/bri-3/src/bri/ccsm/contracts/Ccsm.sol:12]: lets add a maximum length to keep things reasonable

File: examples/bri-3/src/bri/ccsm/contracts/Ccsm.sol [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    require(bytes(_anchorHash).length > 0, 'AnchorHash cannot be empty');
```
Comment (Therecanbeonlyone1969) [inline examples/bri-3/src/bri/ccsm/contracts/Ccsm.sol:13]: it is a hash so it has a maximum length. lets add that check

**Review by Therecanbeonlyone1969 — COMMENTED** [review]
see comments. once addressed good to merge ... we also need tests for the contract, please, and thank you!

File: examples/bri-3/src/bri/ccsm/contracts/Ccsm.sol (hunk 1 above)
Comment (biscuitdey) [reply examples/bri-3/src/bri/ccsm/contracts/Ccsm.sol]: Changes made.

File: examples/bri-3/src/bri/ccsm/contracts/Ccsm.sol (hunk 2 above)
Comment (biscuitdey) [reply examples/bri-3/src/bri/ccsm/contracts/Ccsm.sol]: Changes made.

File: examples/bri-3/src/bri/ccsm/contracts/Ccsm.sol (hunk 3 above)
Comment (biscuitdey) [reply examples/bri-3/src/bri/ccsm/contracts/Ccsm.sol]: Changes made.

**Review by Therecanbeonlyone1969 — APPROVED** [review]
LGTM

**Pull request merged**